### ISSUE TITLE
How do you add a custom validator and error message?

### ISSUE THREAD

**Issue by Xeoncross** [go-playground/validator#711]
### Package version eg. v9, v10: 

v10

It seems adding a custom validation rule is something that should be pretty common. I've been unable to find a complete example of adding a validation rule and the matching error message. Could someone explain how this is to be done?

https://github.com/go-playground/validator/blob/d6b17fd90bd4dd9d16a594c3035ceadc3de0193a/doc.go#L1273-L1282

```go
	type Test struct {
		TestField string `validate:"yourtag"`
	}
	
	validate := validator.New()
	validate.RegisterValidation("yourtag", validators.NotBlank)
	
	// now how do you set the error message?
```

It looks like you might have to add this [non-intuitive block of code to mutate the translation handling](https://github.com/go-playground/validator/blob/f6584a41c8acc5dfc0b62f7962811f5231c11530/_examples/translations/main.go#L105-L111)

```go
	validate.RegisterTranslation("yourtag", trans, func(ut ut.Translator) error {
		return ut.Add("yourtag", "{0} must have a value!", true) // see universal-translator for details
	}, func(ut ut.Translator, fe validator.FieldError) string {
		t, _ := ut.T("yourtag", fe.Field())
		return t
	})
```

Related

- https://github.com/go-playground/validator/issues/669

**Comment by megamayoy** [comment]
Yes, this is how it's done. For a more clarifying example , refer to this blog post https://medium.com/@apzuk3/input-validation-in-golang-bc24cdec1835
